### PR TITLE
[TH2-916] Table for response, new "No response message received" even…

### PR DESCRIPTION
**Pull request by alexander-zhilov**
…t with response searching parameters and linked messages.

### CODE REVIEW

File: src/main/java/com/exactpro/th2/act/ActHandler.java [hunk 1]
```diff
@@ -15,66 +15,56 @@
...
-import com.exactpro.th2.check1.grpc.Check1Service;
+import com.exactpro.th2.common.event.Event;
+import com.exactpro.th2.common.event.Event.Status;
+import com.exactpro.th2.common.event.IBodyData;
+import com.exactpro.th2.common.event.bean.TreeTable;
+import com.exactpro.th2.common.event.bean.builder.CollectionBuilder;
+import com.exactpro.th2.common.event.bean.builder.MessageBuilder;
+import com.exactpro.th2.common.event.bean.builder.RowBuilder;
+import com.exactpro.th2.common.event.bean.builder.TreeTableBuilder;
+import com.exactpro.th2.common.grpc.*;
```
Comment (RaevskiySV) [inline src/main/java/com/exactpro/th2/act/ActHandler.java:34]: Avoid these kind of imports.
Importing specific classes is better approach.

File: src/main/java/com/exactpro/th2/act/ActHandler.java [hunk 2]
```diff
@@ -297,30 +298,66 @@ private EventID createAndStoreParentEvent(PlaceMessageRequestOrBuilder request,
         }
     }
 
+    private void createAndStoreNoResponseEvent(String actName, Map<String, CheckMetadata> expectedMessages, String fieldValue,
+                                               Instant start,
+                                               EventID parentEventId, List<MessageID> messageIDList) throws JsonProcessingException {
+
+        Event errorEvent = Event.from(start)
```
Comment (RaevskiySV) [inline src/main/java/com/exactpro/th2/act/ActHandler.java:305]: I think that proper indent will provide more readability to the code.

File: src/main/java/com/exactpro/th2/act/ActHandler.java [hunk 3]
```diff
@@ -297,30 +298,66 @@ private EventID createAndStoreParentEvent(PlaceMessageRequestOrBuilder request,
...
+
+    private IBodyData createNoResponseBody(Map<String, CheckMetadata> expectedMessages, String fieldValue) {
+        TreeTableBuilder treeTableBuilder = new TreeTableBuilder();
+        CollectionBuilder passedOn = new CollectionBuilder();
+        CollectionBuilder failedOn = new CollectionBuilder();
+        for (Map.Entry<String, CheckMetadata> entry : expectedMessages.entrySet()) {
+            if (entry.getValue().eventStatus == PASSED) {
+                passedOn.row(entry.getKey(), new CollectionBuilder().row(entry.getValue().fieldName, new RowBuilder().column(new EventUtils.MessageTableColumn(fieldValue)).build()).build());
+            }
+            if (entry.getValue().eventStatus == FAILED) {
```
Comment (RaevskiySV) [inline src/main/java/com/exactpro/th2/act/ActHandler.java:325]: Without using else we always double check the eventStatus. This affects efficiency in terms of time complexity.

**Review by RaevskiySV — APPROVED** [review]

**Review by RaevskiySV — APPROVED** [review]

File: src/main/java/com/exactpro/th2/act/ActHandler.java [hunk 4]
```diff
@@ -97,6 +98,21 @@
...
+
+    private static Timestamp getTimestamp(Instant instant) {
+        return Timestamp.newBuilder()
+                .setSeconds(instant.getEpochSecond())
+                .setNanos(instant.getNano())
+                .build();
+    }
+
+    private static String toDebugMessage(com.google.protobuf.MessageOrBuilder messageOrBuilder) throws InvalidProtocolBufferException {
+        return JsonFormat.printer().omittingInsignificantWhitespace().print(messageOrBuilder);
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/act/ActHandler.java:113]: Maybe use the TextFormat.shortDebugString method

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**